scanpy/tests/test_get.py

```diff
@@ -27,22 +31,66 @@ def test_obs_df(adata):
         X=np.zeros((2, 2)),
         var=pd.DataFrame({"gene_symbols": ["raw1", "raw2"]}, index=["gene1", "gene2"]),
     )
-    assert np.all(np.equal(
-        sc.get.obs_df(adata, keys=["gene2", "obs1"], obsm_keys=[("eye", 0), ("sparse", 1)]),
-        pd.DataFrame({"gene2": [1, 1], "obs1": [0, 1], "eye-0": [1, 0], "sparse-1": [0, 1]}, index=adata.obs_names),
-    ))
-    assert np.all(np.equal(
-        sc.get.obs_df(adata, keys=["genesymbol2", "obs1"], obsm_keys=[("eye", 0), ("sparse", 1)], gene_symbols="gene_symbols"),
-        pd.DataFrame({"genesymbol2": [1, 1], "obs1": [0, 1], "eye-0": [1, 0], "sparse-1": [0, 1]}, index=adata.obs_names),
-    ))
-    assert np.all(np.equal(
-        sc.get.obs_df(adata, keys=["gene2", "obs1"], layer="double"),
-        pd.DataFrame({"gene2": [2, 2], "obs1": [0, 1]}, index=adata.obs_names),
-    ))
-    assert np.all(np.equal(
-        sc.get.obs_df(adata, keys=["raw2", "obs1"], gene_symbols="gene_symbols", use_raw=True),
-        pd.DataFrame({"raw2": [0, 0], "obs1": [0, 1]}, index=adata.obs_names),
-    ))
+    assert np.all(
```

Comment:
Looking at this again, I think `pd.testing.assert_frame_equal` should be used here. I believe it will give more helpful error messages.

Reply:
Done. Here I had some trouble with the dtype pandas testing now checks. I now set the type when creating the Anndata object specifically.